File: backend/src/main/java/pl/sda/orange/java2/event/model/Event.java
package pl.sda.orange.java2.event.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "events")
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private LocalDate startDate;
    private LocalDate endDate;
    private String description;


    public Event(String title, LocalDate startDate, LocalDate endDate, String description) {
        this.title = title;
        this.startDate = startDate;
        this.endDate = endDate;
        this.description = description;
    }
}

File: frontend/Event-Managment-Service-Frontend/src/app/model/event.ts
import { User } from './user';

export interface Event {
  id: number;
  title: string;
  startDate: Date;
  endDate: Date;
  description: string;
  // host: User;
}

export interface EventData {
  title: string;
  startDate: Date;
  endDate: Date;
  description: string;
  // host: User;
}
